lib/ansible/modules/cloud/amazon/aws_kms.py

```diff
@@ -915,13 +916,61 @@ def assert_policy_shape(policy):
             if statement['Sid'] == sidlabel:
                 found_statement_type[label] = True
 
-    for statementtype in statement_label.keys():
+    for statementtype in statement_label:
         if not found_statement_type.get(statementtype):
             errors.append('Policy is missing {0}.'.format(statementtype))
 
-    if len(errors):
-        raise Exception('Problems asserting policy shape. Cowardly refusing to modify it: {0}'.format(' '.join(errors)) + "\n" + str(policy))
-    return None
+    if errors:
+        module.fail_json(msg='Problems asserting policy shape. Cowardly refusing to modify it', errors=errors, policy=policy)
+
+
+def canonicalize_alias_name(alias):
+    if alias is None:
+        return None
+    if alias.startswith('alias/'):
+        return alias
+    return 'alias/' + alias
+
+
+def fetch_key_metadata(connection, module, key_id, alias):
+
+    alias = canonicalize_alias_name(module.params.get('alias'))
+
+    # Fetch by key_id where possible
+    if key_id:
+        try:
+            return get_kms_metadata_with_backoff(connection, key_id)['KeyMetadata']
+        except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError):
+            return None
```

Comment:
This was a pre-existing bug, but the exception handling for these would be better like
```python
except is_boto3_error_code('NotFoundException'):
    # Key does not exist
    return None
except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError):  # pylint: disable=duplicate-except
    # Legitimate failure
    module.fail_json_aws(...
```
Right now any actual issues are ignored

Reply:
@s-hertel using client exceptions rather than botocore exceptions is nicer when they actually exist (which they do here for a kms client):
```
except connection.exceptions.NotFoundException:
     return None
except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
    # Legitimate failure
    module.fail_json_aws(e, ...
```

Reply:
I didn't know either of those existed.  Much nicer!

Reply:
@willthames That's definitely nicer!